Allow different clipping on InsetIndicator rectangle and connectors.

juseg opened the pull request:
## PR summary

This adds `clip_on` to the tuple of `_shared_properties` in `matplotlib.inset` so that `InsetIndicator` recognizes that its `rectangle` and `connectors` components should be drawn independently if they have different `clip_on` values (closes #30642, see that issue for a minimal example). 

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

juseg committed:
Add clip_on to matplotlib.inset._shared_properties.

juseg committed:
Add InsetIndicator.set_clip_on().

rcomer commented:
Hi @juseg are you still interested in working on this?  Do you need any more help?

juseg commented:
Hi @rcomer thanks for the heads-up and sorry for letting this sleep, I am back at it will try and add tests based on the advice above!

juseg committed:
Merge remote-tracking branch 'upstream/main' into inset-clipping

juseg committed:
Update inset indicator stub file.

juseg committed:
Test clipping a single connector to axes bbox.

juseg committed:
Update the corresponding baseline image.

juseg committed:
Test updating rect and connector clipping (fails).

juseg committed:
Propagate clip_on to new connectors (still fails).

juseg committed:
The attribute name is _clipon not _clip_on (pass).

juseg committed:
Address ruff complaints.

juseg committed:
Add example of an inset across broken axes.

juseg committed:
Add new example to subplots gallery.

juseg committed:
Merge remote-tracking branch 'upstream/main' into inset-clipping

juseg commented:
- I reworked the image comparison test as suggested.
- Amended another test on rectangle and connector properties.
- I added an example using `clip_on=True` on the rectangle but not the connectors to draw an inset indicator over broken axes (this is actually my original use case for wanting this implemented).

I am happy to get feedback at this stage. Specifically:

- Is this considered "_new features and API changes_" (see PR checklist)?
- Do we want `clip_on=True` by default on the rectangle, while keeping it `False` on the connectors (it would seem like a logical choice to me, but also a breaking change)?

_Edit_: @melissawm Github Actions seems to remove that `topic: axes` tag with every commit.

melissawm commented:
Huh! Looks like a bug in our labeler action - I'll take a look. Thanks for flagging @juseg !
